src/bedrock/1.3/SubChunk.js

```diff
@@ -47,6 +48,36 @@ class SubChunk {
     let storageCount = 1
 
     switch (this.subChunkVersion) {
+      case 0: {
+        const blockIds = stream.readBuffer(4096)
+        const metas = stream.readBuffer(2048)
+
+        for (let x = 0; x < 16; x++) {
+          for (let y = 0; y < 16; y++) {
+            for (let z = 0; z < 16; z++) {
+              const index = (x << 8) + (z << 4) + y
+              const id = blockIds[index]
+              const meta = metas[index >> 1] >> (index & 1) * 4 & 15
+
+              const str = legacyBlockIdMap[`${id}:${meta}`].substring(10)
+              const name = str.split('[', 1)[0]
+              let propertiesArr = []
+              if (str.slice(name.length + 1, -1) !== '') {
+                propertiesArr = str.slice(name.length + 1, -1).split(',')
+              }
+              const properties = Object.fromEntries(propertiesArr.map(property => {
+                const [key, value] = property.split('=')
+                const intValue = parseInt(value)
+                return [key, isNaN(intValue) ? { true: 1, false: 0 }[value] ?? value : intValue]
+              }))
+              const b = this.Block.fromProperties(name, properties, 0)
+              this.setBlock(undefined, x, y, z, b)
```

Comment:
Doing all this logic on every new block inside a subchunk is a very expensive operation. Instead, you can add to the top a mapping for ID + index => blockStateID.

```js
const legacyBlockIdToNewStateId = Object.fromEntries(Object.entries(legacyBlockIdMap).map( ([k, v]) => [k, Block.fromString(v).stateId] ))
```

Then when indexing,
```js
const stateId = legacyBlockIdToNewStateId[id + ':' + meta] ?? legacyBlockIdToNewStateId[id + ':0']
this.setBlockStateId(0, x, y, z, stateId)
```